### PR TITLE
Fix data array shapes for CoDICE-Lo products

### PR DESCRIPTION
**Pull request by bourque**
This PR fixes some issues with how CoDICE-Lo data arrays are shaped, as well as an indexing issue in Lossy A and B table lookup.

### CODE REVIEW

File: imap_processing/codice/codice_l1a.py [hunk 1]
```diff
@@ -129,12 +129,12 @@ def define_coordinates(self) -> None:
...
             if name == "epoch":
                 values = self.dataset.epoch.data
+            elif name == "esa_step":
+                values = np.arange(self.config["num_energy_steps"])
             elif name == "inst_az":
                 values = np.arange(self.config["num_positions"])
             elif name == "spin_sector":
                 values = np.arange(self.config["num_spin_sectors"])
-            elif name == "esa_step":
-                values = np.arange(self.config["num_energy_steps"])
```
Comment (bourque) [inline imap_processing/codice/codice_l1a.py:137]: I just changed the ordering here for consistency with the order of the array shapes

File: imap_processing/codice/decompress.py [hunk 2]
```diff
@@ -50,7 +50,9 @@ def _apply_lossy_a(compressed_bytes: bytes) -> list[int]:
         The 24- or 32-bit decompressed values.
     """
     compressed_values = list(compressed_bytes)
-    decompressed_values = [LOSSY_A_TABLE[item] for item in compressed_values]
+    decompressed_values = [
+        LOSSY_A_TABLE[item - 1] if item > 0 else 0 for item in compressed_values
+    ]
```
Comment (tech3371) [inline imap_processing/codice/decompress.py:55]: Why decompressed value is value of `item - 1` meaning the previous value?

**Review by tech3371 — COMMENTED** [review]
I had one clarification question. Besides, LGTM

File: imap_processing/codice/decompress.py (hunk 2 above)
Comment (bourque) [reply imap_processing/codice/decompress.py:55]: I am using it as a way to implement the "<" condition for the lookup table. From Greg Dunn's instructions on this:

Screenshot 2024-12-20 at 15 49 25

So here I choose the previous index that corresponds to the value.

**Review by tech3371 — APPROVED** [review]

**Pull request merged**